Update from Weblate.

xwikiorgci opened the pull request.

vmassol commented on application-gardening-ui/src/main/resources/Gardening/Code/GardeningTranslations.fr.xml:
```diff
@@ -0,0 +1,83 @@
+gardening.job.success=Terminé !
+gardening.job.error=Une erreur est survenue, consultez les journaux pour plus d'information.
+Gardening.Code.GardeningScriptConfigurationClass_activeQueryScripts=Scripts de requête actifs
+Gardening.Code.GardeningScriptConfigurationClass_activeActionScript=Script d'action actif</content>
```
is there a problem of lowercase/uppercase. The app name is not consistent. It should be the lowercase version which is good.

vmassol replied:
See also http://dev.xwiki.org/xwiki/bin/view/Community/L10N%20Conventions

aubincleme replied:
IMO it's more a problem of the L10N practices that are not consistent with the reality of contrib applications. 

Here are two examples : 
* https://github.com/xwiki-contrib/application-blog/tree/master/application-blog-ui/src/main/resources/Blog
* https://github.com/xwiki-contrib/application-forum/tree/master/application-forum-ui/src/main/resources

vmassol replied:
> IMO it's more a problem of the L10N practices that are not consistent with the reality of contrib applications.

Can you explain what you mean? 

And why it's a problem specific to contrib apps?

Are you also suggesting that it's a good idea to have "gardening" and "Gardening" prefixes for the same module for example?

Thanks

vmassol replied:
Example 1 (bad but historic probably): https://github.com/xwiki/xwiki-platform/blob/11b5d1a756ee36b6d47fe9d22dfa7b6c662d3c81/xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/UserProfileTranslations.xml

Example2 (good): https://github.com/xwiki/xwiki-platform/blob/beb480316ad38107fb980529d2efcb103cea7539/xwiki-platform-core/xwiki-platform-activitystream/xwiki-platform-activitystream-ui/src/main/resources/Main/ActivityTranslations.xml

aubincleme replied:
So here, `Gardening` keys (with capital letters) refers to page names and are used to translate class properties. 
Personally, I've never seen a wiki page that belongs to an application and that starts with a non-capital letter.

Moreover, the two examples I gave above (https://github.com/xwiki-contrib/application-gardening/pull/1#discussion_r188912541) show applications that uses capital letters in their translation keys (see https://github.com/xwiki-contrib/application-forum/blob/f969f31003522febd88bc6a2a0b45f232c2cbd34/application-forum-ui/src/main/resources/ForumCode/Translations.xml#L43).

vmassol replied:
Can you tell me where we have a rule that says that a translation needs to use the wiki page name for the prefix? (I don't recall this).

Your example 1 link for the forum is bad since that clearly doesn't follow the best practices.

vmassol replied:
I vaguely remember some naming rules that we need to follow for ConfigurableClass but I don't recall the details.

Maybe you can explain the context where you use your translations with uppercases?

vmassol replied:
Found one here: https://github.com/xwiki-contrib/application-gardening/blob/f2135107f13e8691ba495d4ba056311943859311/application-gardening-ui/src/main/resources/Gardening/Code/GardeningScriptAdministration.xml#L61 

This is clearly not following any best practice I know :) But maybe I'm missing something so any link would help.

vmassol replied:
In addition it's really a bad practice IMO since it ties a translation to a wiki page location which will break as soon as you refactor the page and move code to another location...

aubincleme replied:
So, just to be sure about the problems we're having now, you don't think that translation keys used for translating class properties should have prefixes with capital letters ?

aubincleme replied:
We have a ton of examples like that in platform, here are two other ones : 
* https://github.com/xwiki/xwiki-platform/blob/fa832612e6025bca86a3c00e1e2667fe20747131/xwiki-platform-core/xwiki-platform-invitation/xwiki-platform-invitation-ui/src/main/resources/Invitation/Translations.xml#L45
* https://github.com/xwiki/xwiki-platform/blob/fa832612e6025bca86a3c00e1e2667fe20747131/xwiki-platform-core/xwiki-platform-annotations/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Translations.xml#L110

tmortagne replied:
@vmassol this does not have much to do with naming best practice, that's just how class property translations works. When you want to provide a translation for a class property you have to follow a specific syntax `<class reference>_<property name>` and the reference of the class here is `Gardening.Code.GardeningScriptConfigurationClass`. If you don't you won't see it in the object editor. See http://dev.xwiki.org/xwiki/bin/view/Community/L10N%20Conventions#HTranslatingXClasses.

vmassol replied:
@tmortagne Yes you're right for class property translations but in the case of @aubincleme this is not the case. See the link I posted above: https://github.com/xwiki-contrib/application-gardening/blob/f2135107f13e8691ba495d4ba056311943859311/application-gardening-ui/src/main/resources/Gardening/Code/GardeningScriptAdministration.xml#L61

tmortagne replied:
To me it just mean this code reuse the standard activeQueryScripts class property translation. I don't see the point in providing two.

vmassol replied:
ok I see what you mean. I didn't realize this translation key was used for different places.

vmassol replied:
Ok so now the next step to discuss (not related to the gardening app) and long term is whether we think it's a good idea to have translation keys named after a page (especially since that makes it hard to refactor the page). BTW just rearealizing that we may have broken lots of translations (and still are) when we moved code pages under the Code space...

FTR, the LT solves this by offering a translation prefix that can be specified by the dev.

I'll bring this to the devs list since here is not the right place to discuss it :)

vmassol replied:
Posted to https://markmail.org/message/wu7qkuu7mevel5qp

aubincleme replied:
Ok, as this problem is resolved for now and for this issue ; merging …

aubincleme committed:
Translated using Weblate (French)

Currently translated at 100.0% (7 of 7 strings)

Translation: XWiki Contrib/Application Gardening
Translate-URL: https://l10n.xwiki.org/projects/xwiki-contrib/application-gardening/fr/
